mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/search/ElasticSearchSearcher.java

```diff
@@ -82,18 +87,24 @@ public ElasticSearchSearcher(Client client, QueryConverter queryConverter, int s
         this.typeName = typeName;
     }
 
-    public Stream<MessageSearchIndex.SearchResult> search(Collection<MailboxId> mailboxIds, SearchQuery query,
+    public SearchResult search(Collection<MailboxId> mailboxIds, SearchQuery query,
                                                           Optional<Long> limit) throws MailboxException {
         SearchRequestBuilder searchRequestBuilder = getSearchRequestBuilder(client, mailboxIds, query, limit);
+        Cardinality cardinality = searchRequestBuilder.get()
+            .getAggregations()
+            .get(CARDINALITY_ARGUMENT);
+
         Stream<MessageSearchIndex.SearchResult> pairStream = new ScrollIterable(client, searchRequestBuilder).stream()
             .flatMap(this::transformResponseToUidStream);
 
-        return limit.map(pairStream::limit)
-            .orElse(pairStream);
+        return new SearchResult(cardinality.getValue(), limit.map(pairStream::limit)
+            .orElse(pairStream));
     }
 
     private SearchRequestBuilder getSearchRequestBuilder(Client client, Collection<MailboxId> users,
                                                          SearchQuery query, Optional<Long> limit) {
+        CardinalityBuilder aggregation = AggregationBuilders.cardinality(CARDINALITY_ARGUMENT)
```

Comment:
I understand why it is needed but that is a bit of a shame.

Maybe some strong sign we need to think about denormalization in ElasticSearch.

CF @mbaechler  @aduprat  @rouazana 

Of course IMO not to fix in this PR, but it will have some performance implication.

(BTW we need it because documents are indexed by mailboxId+uid but we want unique messageId)